### DISCUSSION TITLE
Use secrets in a public repository

### DISCUSSION THREAD

**Discussion by ianfirme** [community/community#59690] [Actions]
### Select Topic Area

Question

### Body

how can i use secrets in a public repository and ensure that they remain protected and not visible in github actions log?

**Comment by tcortega** [comment]
Secrets are encrypted environment variables created in a repository or organization for use in GitHub Actions workflows. They are not passed to workflows that are triggered by a pull request from a fork. GitHub automatically redacts secrets printed to the log, but you should avoid printing secrets to the log intentionally. Note that usage of secrets in public repositories should be done cautiously as they can be potentially exposed by malicious PRs.

**Reply by Maher27277** [reply]
<details><summary>Details</summary>
<p>



</p>
</details>
